cli/clitest/clitest.go

```diff
@@ -155,6 +159,13 @@ func Start(t *testing.T, inv *clibase.Invocation) {
 	go func() {
 		defer close(closeCh)
 		err := waiter.Wait()
+
+		if errExpected {
+			assert.Error(t, err)
+			assert.False(t, errors.Is(err, context.Canceled), "error was expected, but context was canceled")
+			return
+		}
```

Comment:
From the function signature, I'd expect this to take a func() that lets you pass in any assertions you want?

Not a blocker, but would make this very flexible and useful.

Reply:
Something like `assertCallback func(t)`?

Reply:
Yeah something like that!